src/Network/Algorand/Util.hs

```diff
@@ -108,11 +132,31 @@ lookupAppLocalState Api.Account{..} appId = do
   M.fromList . map toEntry <$> lsKeyValue
   where
     toPair a@Api.LocalState{..} = (lsId, a)
-    toEntry Api.TealKeyValue{..}
-      | Api.tvType tkeValue == Api.tealValueBytesType
-      = (tkeKey, Left $ Api.tvBytes tkeValue)
-      | Api.tvType tkeValue == Api.tealValueUintType
-      = (tkeKey, Right $ Api.tvUint tkeValue)
+    toEntry TealKeyValue{..}
+      | tvType tkeValue == tealValueBytesType
+      = (tkeKey, Left $ tvBytes tkeValue)
+      | tvType tkeValue == tealValueUintType
+      = (tkeKey, Right $ tvUint tkeValue)
       | otherwise = error $
         "lookupAppLocalState: unknown teal value type "
-        <> show (Api.tvType tkeValue)
+        <> show (tvType tkeValue)
+
+-- | Helper to get account local state from indexer
+getAppLocalState
+  :: Api.AccountData
+  -> AppIndex
+  -> Maybe (Map ByteString (Either ByteString Word64))
+getAppLocalState Api.AccountData{..} appId = do
```

Comment:
Can you please extract common logic between `lookupAppLocalState` and `getAppLocalState` into a separate function?

Reply:
Yes, I can. But my logic was, that in future we are going to delete `lookupAppLocalState` (because actually this is same functions, but one for node and other one for indexer)

Reply:
I wanted to write "yes, but we don't know how soon we'll delete `lookupAppLocalState`, this duplication may cause some troubles before we delete deprecated stuff", but then I realized that nobody except for us really uses this library, so we can delete deprecated stuff pretty soon. So ok, let's not care about this duplication.